Applications/WebPortal/WebApp/library/source/pages/account_page/compliance_page/compliance_page.tsx

```diff
@@ -43,36 +51,57 @@ export class CompliancePage extends React.Component<Properties, State> {
       <div style={CompliancePage.STYLE.wrapper}>
         <div style={CompliancePage.STYLE.filler}/>
         <div style={contentStyle}>
-          <RulesList 
+          <RulesList
             displaySize={this.props.displaySize}
             currencyDatabase={this.props.currencyDatabase}
             complianceList={this.state.complianceList}
-            onChange={this.onChange}/>
+            onChange={this.onRuleChange}/>
           <div style={CompliancePage.STYLE.paddingMedium}/>
-          <NewRuleButton displaySize={this.props.displaySize}/>
+          <NewRuleButton displaySize={this.props.displaySize}
+            isOpen={this.state.isAddRulewModalOpen}
+            onToggleModal={this.onToggleAddRuleModal}
+            onAddNewRule={this.onAddNewRule}
+            ruleSchemas={this.props.ruleSchemas}/>
           <div style={CompliancePage.STYLE.paddingLarge}/>
         </div>
         <div style={CompliancePage.STYLE.filler}/>
       </div>);
   }
 
-  private onChange(ruleIndex: number, newRule: Nexus.ComplianceRuleEntry) {
+  private onRuleChange(ruleIndex: number, newRule: Nexus.ComplianceRuleEntry) {
     this.state.complianceList[ruleIndex] = newRule;
     this.setState({complianceList: this.state.complianceList});
   }
 
+  private onToggleAddRuleModal() {
+    this.setState({isAddRulewModalOpen: !this.state.isAddRulewModalOpen});
+  }
+
+  private onAddNewRule(schema: Nexus.ComplianceRuleSchema) {
+    this.state.complianceList.push(
+      new Nexus.ComplianceRuleEntry(
+        0,
+        Beam.DirectoryEntry.makeDirectory(0, 'empty'),
+        Nexus.ComplianceRuleEntry.State.DISABLED,
+        schema
+      )
+    );
+    this.setState({complianceList: this.state.complianceList});
+  }
+
   private static readonly STYLE = {
     wrapper: {
       paddingTop: '18px',
       paddingLeft: '18px',
       paddingRight: '18px',
       paddingBottom: '60px',
+      height: '100vh',
```

Comment:
You are most likely going to have to remove this height. This page isn't a "stand-alone" page but rather is a sub-page that will be inserted into a parent page.

Reply:
Thanks for reminding me to remove it. I thought I caught all the relics of a earlier experiment.